### PR TITLE
Introduce `reset_{at|after}` functions for ticker.

### PR DESCRIPTION
**Pull request by Frostie314159**
These functions perform a similar job to reset, by reseting the timer either after a specified duration or at an instant.
The old PR was closed, since I renamed the branch to be more descriptive.

### CODE REVIEW

**Review by lulf — APPROVED** [review]
LGTM, thanks!

**Comment by Frostie314159** [comment]
I think the CI needs to be restarted, since three tests are still waiting for status.

**Comment by lulf** [comment]
bender run

**Comment by lulf** [comment]
@Frostie314159 Could you try to rebase this branch please?

File: embassy-time/src/timer.rs [hunk 1]
```diff
@@ -190,6 +190,18 @@ impl Ticker {
         self.expires_at = Instant::now() + self.duration;
     }
 
+    /// Reset the ticker to fire for the next time on the deadline.
+    /// If the deadline is in the past, the ticker will fire instantly.
+    pub fn reset_at(&mut self, deadline: Instant) {
+        self.expires_at = deadline;
```
Comment (Dirbaio) [inline embassy-time/src/timer.rs:196]: this is inconsistent with `reset` and `reset_after`. 

`reset` resets it at "now", so it expires next at "now + self.duration"
`reset_after` resets it at "now + after", so it expires next at "now + after + self.duration"

so if `reset_at` resets it at "deadline", it should expire next at "deadline + self.duration"

Comment (Frostie314159) [reply embassy-time/src/timer.rs:196]: I've updated it accordingly.

Comment (Dirbaio) [reply embassy-time/src/timer.rs:196]: can't see the change, maybe you didn't push?

Comment (Frostie314159) [reply embassy-time/src/timer.rs]: I pushed it to the wrong branch. After fighting with git for a while, it should've worked now.

Comment (Frostie314159) [reply embassy-time/src/timer.rs]: I pushed it to the wrong branch. After fighting with git for a while, it should've worked now.

**Review by Dirbaio — APPROVED** [review]

**Pull request merged**